src/app/components/ClientConfigLoader.tsx

```diff
@@ -4,9 +4,26 @@ import { ClientConfig } from '../hooks/useClientConfig';
 import { trimTrailingSlash } from '../utils/common';
 
 const getClientConfig = async (): Promise<ClientConfig> => {
-  const url = `${trimTrailingSlash(import.meta.env.BASE_URL)}/config.json`;
-  const config = await fetch(url, { method: 'GET' });
-  return config.json();
+  let url = `${trimTrailingSlash(import.meta.env.BASE_URL)}/config.${window.location.hostname}.json`;
+  let config = await fetch(url, { method: 'GET' });
+
+  let loadedConfig = null;
+  try {
+    if(config.ok && import.meta.env.MODE != "development") {
+      loadedConfig = await config.json();
+    }
+  } catch (e) {}
+
+  if(!loadedConfig) {
+    url = `${trimTrailingSlash(import.meta.env.BASE_URL)}/config.json`;
+    config = await fetch(url, { method: 'GET' });
+    try {
+      loadedConfig = await config.json();
+    } catch (e) {
+      return {};
+    }
+  }
```

Comment:
Please do not catch (and suppress) the error here as UI is responsible for caching and displaying them to user.